app/recipe/views.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework import (
+    viewsets,
+    status,
+)
+from rest_framework.response import Response
+
+from core.recipe_models import Recipe
+from recipe import serializers
+
+
+class RecipeViewSet(viewsets.ModelViewSet):
+    serializer_class = serializers.RecipeDetailSerializer
+    queryset = Recipe.objects.all()
+
+    def get_queryset(self):
+        return self.queryset.order_by('-id')
```

Comment:
You could do something like this:
```suggestion
    def get_queryset(self):
        queryset = self.queryset.order_by('-id')
        name = self.request.query_params.get('name')
        if name is not None:
            queryset = queryset.filter(name__icontains=name)
        return queryset
```

I don't remember the exercise so well, but if I'm not mistaken the name filter should be only for applicable for the `GET /recipies` endpoint. To achieve that you could split the viewset in two or more, one for list and other for details.